Pinta.Core/Extensions/OtherExtensions.cs

```diff
@@ -9,11 +9,38 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Collections.ObjectModel;
 
 namespace Pinta.Core
 {
 	public static class OtherExtensions
 	{
+		/// <returns>Collection wrapper backed by immutable collection</returns>
+		public static ReadOnlyCollection<T> ToReadOnlyCollection <T> (this IEnumerable<T> values)
+		{
+			switch (values) {
+				case ImmutableBackedReadOnlyCollection<T> transparent:
+					return transparent;
+				case ImmutableArray<T> array:
+					return new ImmutableBackedReadOnlyCollection<T> (array);
+				case ImmutableList<T> list:
+					return new ImmutableBackedReadOnlyCollection<T> (list);
+				default:
+					return new ImmutableBackedReadOnlyCollection<T> (values.ToImmutableArray ());
```

Comment:
In the `FormatDescriptor` use case, there aren't any methods which will modify the lists so I feel like just having a plain `ReadOnlyCollection` wrapping the list is probably fine?
I'm not sure I see the need for doing an extra copy into an immutable array